### ISSUE TITLE
Trait implementation incorrectly shows calls as recursive in some scenarios

### ISSUE THREAD

**Issue by bgourlie** [intellij-rust/intellij-rust#987]
Consider the following:

```rust
fn main() {
    trait ThingTrait {
        fn inc(&mut self) -> u8;
        fn dec(&mut self) -> u8;
    }

    struct Thing {
        val: u8
    }
    
    impl Thing {
        fn inc(&mut self) -> u8 {
            self.val += 1;
            self.val
        } 
    }
    
    impl ThingTrait for Thing {
        fn inc(&mut self) -> u8 {
            self.inc() // <-- Incorrectly marked as recursive, calls into struct impl
        }
        
        fn dec(&mut self) -> u8 {
            self.dec() // <-- Correctly marked as recursive (and a compiler warning)
        }
    }
    
    let mut thing = Thing { val: 0 };
    println!("Struct impl call: {}, Trait impl call: {}", thing.inc(), ThingTrait::inc(&mut thing))
}
```

**Comment by alygin** [comment]
I believe it's fixed in #967. The fix should make it to alpha today.

**Comment by matklad** [comment]
Yep, it is fixed!
